tests/gocase/unit/protocol/regression_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package protocol
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/apache/incubator-kvrocks/tests/gocase/util"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRegression(t *testing.T) {
+	srv, err := util.StartServer(t, map[string]string{})
+	require.NoError(t, err)
+	defer srv.Close()
+
+	ctx := context.Background()
+	rdb := srv.NewClient()
+	defer func() { require.NoError(t, rdb.Close()) }()
+
+	c := srv.NewTcpClient()
+	defer func() { require.NoError(t, c.Close()) }()
+
+	proto := "*3\r\n$5\r\nBLPOP\r\n$6\r\nhandle\r\n$1\r\n0\r\n"
+	require.NoError(t, c.Write(fmt.Sprintf("%s%s", proto, proto)))
+
+	resList := []string{"*2", "$6", "handle", "$1", "a"}
+
+	v := rdb.RPush(ctx, "handle", "a")
+	require.EqualValues(t, 1, v.Val())
+	for _, res := range resList {
+		r, err := c.ReadLine()
+		require.NoError(t, err)
+		require.Equal(t, res, r)
+	}
+
+	v = rdb.RPush(ctx, "handle", "a")
+	require.EqualValues(t, 1, v.Val())
+
+	// TODO should read the second pushed element
+	//for _, res := range resList {
+	//	r, err := c.ReadLine()
+	//	require.NoError(t, err)
+	//	require.Equal(t, res, r)
+	//}
```

Comment:
This is not a regression because upstream tests only the equivalent to:

```go
require.EqualValues(t, 1, v.Val())
```

You can see this comment https://github.com/redis/redis/commit/4bb1ccc6ebaa5944f8283500d69cae505f9d04d8#r83003244.

If I send `BLPOP` commands after `RPUSH`, the reader can read both two responses.

I don't know whether it's a bug/incompatibility in Kvrocks, but it's not a blocker to this patch.